extra/extra_01_intersection_of_two_linked_lists

```diff
@@ -0,0 +1,19 @@
+# Definition for singly-linked list.
+# class ListNode:
+#     def __init__(self, x):
+#         self.val = x
+#         self.next = None
+
+# 時間計算量 O(n)、空間計算量 O(n)
+# Time Limit Exceeded
+class Step1:
+    def getIntersectionNode(self, headA: ListNode, headB: ListNode) -> Optional[ListNode]:
+        appeared = set()
+        while headA:
+            appeared.add(headA)
+            headA = headA.next
+        while headB:
+            if headB in appeared:
```

Comment:
Here headB is checked whether it is in appeared, and this require O(m) where m is the number of nodes of listA.
while loop is required O(n) where n is the number of nodes of listB.
So the total time complexity here is O(m * n).

Reply:
The type of appeared is set, not list.
So, I think set requires O(1) to check headB?
I referred to this page.
[https://qiita.com/Hironsan/items/68161ee16b1c9d7b25fb](url)